File: python-pr/52/python/code-quality-readability-maintainability/data_validator_32.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
def validate_user_data(user_data):
    if not isinstance(user_data, dict):
        return False
    
    required_fields = ['name', 'age']
    for field in required_fields:
        if field not in user_data:
            return False
    
    name = user_data['name']
    age = user_data['age']
    
    if not isinstance(name, str) or len(name) < 2:
        return False
    
    if not isinstance(age, int) or not (0 <= age <= 150):
        return False
    
    return True
# {/fact}

File: js-pr/56/javascript/code-quality-readability-maintainability/data_validator_32.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
function validateUserData(userData) {
    if (!userData || typeof userData !== 'object') {
        return false;
    }
    
    const requiredFields = ['name', 'age'];
    const hasAllFields = requiredFields.every(field => field in userData);
    
    if (!hasAllFields) {
        return false;
    }
    
    const { name, age } = userData;
    
    const isValidName = typeof name === 'string' && name.length >= 2;
    const isValidAge = typeof age === 'number' && age >= 0 && age <= 150;
    
    return isValidName && isValidAge;
}
// {/fact}